### PR TITLE
LinalgRefactor - Remove old linalg::scale

### PR DESCRIPTION
**Pull request by OXPHOS**
Based on #3358

### CODE REVIEW

File: cmake/external/GoogleTestNMock.cmake [hunk 1]
```diff
@@ -8,8 +8,8 @@ SET (GMOCK_REVISION 443)
 include(ExternalProject)
 ExternalProject_Add(
 	GoogleMock
-	URL http://googlemock.googlecode.com/files/gmock-1.7.0.zip
-	URL_MD5 073b984d8798ea1594f5e44d85b20d66
+	URL https://github.com/google/googletest/archive/release-1.8.0.tar.gz
```
Comment (karlnapf) [inline cmake/external/GoogleTestNMock.cmake]: please cherry pick the google mock, separate patch for that, before all the others

**Comment by OXPHOS** [comment]
@karlnapf I am thinking so. Also some operations like `sum` are repeatedly defined in `SGVector` and some ML Algorithms called `SGVector::sum()`. Is it necessary to replace these ones gradually?

**Comment by karlnapf** [comment]
Yes, that will be done in a single step:

 1.  remove all math methods from `SGVector`
 2. Lots of compile errors
 3. Replace the calls with new linalg

So you can always get a feeling for what is missing by doing step 1

**Comment by karlnapf** [comment]
I wouldnt remove things for now. Let's finish the linalg branch, remove old linalg in there, replace calls, merge to develop. Before we touch SGVector

**Comment by OXPHOS** [comment]
@karlnapf This should be ready

**Comment by karlnapf** [comment]
Needs rebase

**Comment by OXPHOS** [comment]
@karlnapf  ready

**Pull request merged**